src/pass/loop_partition.cc

```diff
@@ -513,17 +513,19 @@ Stmt LoopPartitioner::TryPartition(const Node* node,
   bool pre_stmt_recurse = true;
   if (middle_interval_i->HasLowerBound()) {
     body_begin = ir::Simplify(middle_interval.min());
-    Expr cond = (body_begin - min >= 0);
```

Comment:
we can also think about changing this condition to `body_begin - min > 0` instead of using `>=` that can also prevent recursing on zero extent loops.

Reply:
@umangyadav, we can do this one, however we still generate zero extent loop. Just that we dont recursively partition anything generated for zero extent loop. 
I still prefer to generate cleaner output that has no such zero extent loops that need fixing afterwards, unless somehow we fix it in the loop partition itself. This can be done as you suggested via `RemoveNoOp` however it seems little convoluted way to go about this. I would rather have loop partition generate clean IR with valid loop iterations.
Is there a reason why you would prefer to generate zero extent loop?

Reply:
@kimishpatel There is no specific reason or preference for generating the zero extent loops. 

My point is that there is always possibility of having the zero extent loops. 

We need to make sure not to recurse on those loops by having some mechanism e.g either guard conditions, `RemoveNoOp` etc.

Reply:
@umangyadav, I understand. Do you have specific example where we can still generate zero extent loops despite this PR? Then it will be easier to see what would be a better fix.

Reply:
@umangyadav I agree we might need a cleanup pass. But if we can avoid generating zero extent loops, it's desirable.